crates/astria-sequencer-relayer/src/relayer/write/conversion.rs

```diff
@@ -36,23 +38,33 @@ pub(super) struct RollupInfo {
     sequencer_rollup_id: RollupId,
 }
 
-/// Information about the a block that was converted to blobs.
+/// Information about a sequencer block that was converted to blobs.
 #[derive(Debug, serde::Serialize)]
 pub(super) struct ConversionInfo {
     #[serde(serialize_with = "serialize_height")]
     pub(super) sequencer_height: SequencerHeight,
     #[serde(serialize_with = "serialize_namespace")]
     pub(super) sequencer_namespace: Namespace,
-    pub(super) rollups: Vec<RollupInfo>,
+    pub(super) rollups_included: Vec<RollupInfo>,
+    pub(super) rollups_excluded: Vec<RollupInfo>,
 }
 
-/// The result of a block that was converted to blobs.
+/// The result of a sequencer block that was converted to blobs.
 pub(super) struct Converted {
     pub(super) blobs: Vec<Blob>,
     pub(super) info: ConversionInfo,
 }
 
-pub(super) fn convert(block: SequencerBlock) -> eyre::Result<Converted> {
+/// Convert the given sequencer block into a collection of blobs and related metadata.
+///
+/// If `include` is empty, blobs from all rollups will be included.  Otherwise only blobs from the
+/// rollups specified in `include` will be included.
+// allow: we'd need static lifetime on a ref to avoid pass-by-value here.
+#[allow(clippy::needless_pass_by_value)]
+pub(super) fn convert(
+    block: SequencerBlock,
+    include: HashSet<RollupId>,
```

Comment:
I understand the design decision here, but I am worried that this might quickly become confusing a few months down the line.

Instead of passing this `HashSet<RollupId>` around, could this be made into a newtype wrapper `IncludeRollup(HashSet<RollupId>)` with a single `IncludeRollup::include: rollup_id -> bool` that contains this logic?

Extra idea: if we make this `IncludeRollup<Arc<HashSet<RollupId>>` then this can be cheaply cloned (and the `needless_pass_by_value` avoided?).

Reply:
The newtype makes sense to me, but the `Arc` seems a little overkill given I'm expecting the set to be empty or at most have one or two entries, with the entries being `Copy` types.  It's also only being cloned once per sequencer block.  Saying that, I guess the `Arc` can't hurt, so I'll add it :)

The `needless_pass_by_value` still won't be avoided though, since Clippy wants me to change the signature to take a ref to the set, and the lifetime on it would need to be `'static` if I did that.

Reply:
Done in [02b1276](https://github.com/astriaorg/astria/pull/1001/commits/02b1276267c6ac65b72c78191ebecdfae92e72d9).